docs/site/Relations.md

```diff
@@ -0,0 +1,164 @@
+---
+lang: en
+title: 'Relations'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Relations.html
+summary:
+---
+
+## Overview
+
+Model relation in LoopBack 3 is one of its powerful features which help users
+define real-world mappings between their models, access sensible CRUD APIs for
+each of the models, and add querying and filtering capabilities for the relation
+APIs after scaffolding their LoopBack applications. In LoopBack 4, with the
+introduction of [repositories](Repositories.md), we aim to simplify the approach
+to relations by creating constrained repositories. This means that, based on the
+relation definition, certain constraints need to be honoured by the target model
+repository, and thus, we produce a constrained version of it as a navigational
+property on the source repository. The first relation type we have available is
+the `has many` relation which operates by constraining the target repository by
+the foreign key property on its associated model.
+
+## Defining a HasMany Relation
+
+Model relations in LoopBack 4 are defined by having a decorator on a designated
+relational property. The definition of the `has many` relation is done by using
+the `hasMany` decorator. The decorator takes in the target model class
+constructor and optionally a custom foreign key to store the relation metadata.
+The decorator logic also designates the relation type and tries to infer the
+foreign key on the target model (or `keyTo` in the relation metadata) as a
+sensible default value which is used in LoopBack 3. It also calls
```

Comment:
What is the sensible default value here?

Reply:
actually why `sensible`? :)  is it better to omit this word?

Reply:
Sorry if I wasn't clear. I think there should be a concrete formula for how the foreign key is inferred (camelcased target model name + Id)